atlasdb-container-test-utils/src/main/java/com/palantir/atlasdb/containers/CassandraCliParser.java

```diff
@@ -21,17 +21,33 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import com.google.common.base.Preconditions;
+
 public final class CassandraCliParser {
     private static final Logger log = LoggerFactory.getLogger(CassandraCliParser.class);
 
-    private CassandraCliParser() {
+    private static final String REPLICATION_REGEX_2_X = "^.*\"replication_factor\":\"(\\d+)\"\\}$";
+    private static final String REPLICATION_REGEX_3_X = "^.*'replication_factor': '(\\d+)'\\}$";
+
+    private final String cassandraVersion;
+
+    public CassandraCliParser(String cassandraVersion) {
+        Preconditions.checkState(isSupported(cassandraVersion),
+                "Unsupported Cassandra version {} passed to CassandraCliParser",
+                cassandraVersion);
+        this.cassandraVersion = cassandraVersion;
     }
 
-    public static int parseSystemAuthReplicationFromCqlsh(String output) throws IllegalArgumentException {
+    private static boolean isSupported(String cassandraVersion) {
```

Comment:
Static method at the bottom please (makes it easier to see the public methods which are the ones you usually care about)